tests/countries/test_cocos_islands.py

```diff
@@ -0,0 +1,275 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.cocos_islands import CocosIslands, CC, CCK
+from tests.common import CommonCountryTests
+
+
+class TestCocosIslands(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1985, 2050)
+        super().setUpClass(CocosIslands, years=years, years_non_observed=range(1985, 2050))
+        cls.holidays_observed = CocosIslands(observed=True, years=years)
+        cls.holidays_non_observed = CocosIslands(observed=False, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(CocosIslands, CC, CCK)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(CocosIslands(years=1984))
+
+    def test_new_years_day(self):
+        name = "New Year's Day"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1985, 2050)))
+        obs_dt = ("2023-01-02", "2017-01-02", "2028-01-03", "2034-01-02")
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_australia_day(self):
+        name = "Australia Day"
+        self.assertHolidayName(name, (f"{year}-01-26" for year in range(1985, 2050)))
+        obs_dt = ("2020-01-27", "2025-01-27")
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_self_determination_day(self):
+        name = "Act of Self Determination Day"
+        self.assertHolidayName(name, (f"{year}-04-06" for year in range(1985, 2050)))
+        self.assertNoHolidayName(
+            name,
+            (f"{year}-04-06" for year in range(1955, 1984)),
+        )
+        obs_dt = ("2025-04-07",)
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+        self.assertNotIn("1980-04-06", CocosIslands(years=1980))
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(
+            name,
+            [
+                "2020-04-10",
+                "2021-04-02",
+                "2022-04-15",
+                "2023-04-07",
+                "2025-04-18",
+            ],
+        )
```

Comment:
```suggestion
            "2020-04-10",
            "2021-04-02",
            "2022-04-15",
            "2023-04-07",
            "2024-03-29",
            "2025-04-18",
        )
        self.assertHolidayName(name, range(1985, 2050))
```
And the same for similar holidays.